src/nativeMain/kotlin/com/autodesk/coroutineworker/CoroutineWorker.kt

```diff
@@ -150,3 +110,55 @@ actual class CoroutineWorker {
 
     init { freeze() }
 }
+
+private class CoroutineWorkerState {
+
+    /**
+     * The backing store for the state
+     */
+    private val value = AtomicInt(0)
+
+    /**
+     * True, if the job was cancelled; false, otherwise.
+     */
+    var cancelled: Boolean
+        get() = isSet(cancelledBit)
+        set(value) = updateValue(cancelledBit, value)
+
+    /**
+     * True, if the job finished; false, otherwise.
+     */
+    var completed: Boolean
+        get() = isSet(completedBit)
+        set(value) = updateValue(completedBit, value)
+
+    /**
+     * Updates the value with the bit, setting or un-setting it
+     */
+    private fun updateValue(bit: Int, set: Boolean) {
+        value.value = if (set) {
```

Comment:
Nut sure but if this method is called from different threads then there is a race condition here. You first read the current value then change and then save. This operation should be atomic.

Reply:
Great catch! Will fix :)

Reply:
You need something like [this](https://github.com/badoo/Reaktive/blob/93a34ebe23ad3cf3584def7fb30247605ec01fd3/utils/src/commonMain/kotlin/com/badoo/reaktive/utils/atomic/AtomicReferenceExt.kt#L24) but for `AtomicInt`